File: Abstraction/Java/XGames/src/com/example/xgames/Skateboarding.java
package com.example.xgames;

public class Skateboarding extends XGames{
    private String sponsor; //String variable to track of the Skateboarding sponsor

    public Skateboarding(){ //Default constructor
        super();
        this.sponsor = " ";
    }
    public Skateboarding(String sport, String sponsor){ //Create overloaded constructor
        super(sport);
        this.sponsor = sponsor;
    }

    @Override
    public int TopScore(String participant) { //Overridden method to calculate participant's score
        int score = 10 * participant.length();
        return score;
    }
    public void SetSponsor(String sponsor){ //Method to set the Skateboarding sponsor
        this.sponsor = sponsor;
    }
    public String GetSponsor(){ //Method to get the Skateboarding sponsor
        return this.sponsor;
    }
}

File: Abstraction/C++/XGames/Skateboarding.cpp
#include<iostream>
#include"XGames.cpp" //Import the file containing the "Abstract" class
using namespace std;

class Skateboarding:public XGames{
private:
    string sponsor; //String variable to keep track of skateboarding sponsor
public:
    Skateboarding():XGames(){this->sponsor = " ";}//Default constructor
    Skateboarding(string sponsor) : XGames("Skateboarding"){this->sponsor = sponsor;} //Overloaded constructor
    int TopScore(string participant)override{return 10 * participant.size();}; //Calculates the score of the participant
    void SetSponsor(string sponsor){this->sponsor = sponsor;} //sponsor setter
    string GetSponsor(){return this->sponsor;} //sponsor getter
};